src/syncback/mod.rs

```diff
@@ -0,0 +1,495 @@
+        .and_then(|inst| inst.metadata().middleware);
+    let inst = snapshot.new_inst();
+
+    let mut middleware;
+
+    if let Some(override_middleware) = snapshot.middleware {
+        return override_middleware;
+    } else if let Some(old_middleware) = old_middleware {
+        return old_middleware;
+    } else if json_model_classes.contains(inst.class.as_str()) {
+        middleware = Middleware::JsonModel;
+    } else {
+        middleware = match inst.class.as_str() {
+            "Folder" | "Configuration" | "Tool" => Middleware::Dir,
+            "StringValue" => Middleware::Text,
+            "Script" => Middleware::ServerScript,
+            "LocalScript" => Middleware::ClientScript,
+            "ModuleScript" => Middleware::ModuleScript,
+            "LocalizationTable" => Middleware::Csv,
+            // This isn't the ideal way to handle this but it works.
+            name if name.ends_with("Value") => Middleware::JsonModel,
+            _ => Middleware::Rbxm,
+        }
+    }
+
+    if !inst.children().is_empty() {
+        middleware = match middleware {
+            Middleware::ServerScript => Middleware::ServerScriptDir,
+            Middleware::ClientScript => Middleware::ClientScriptDir,
+            Middleware::ModuleScript => Middleware::ModuleScriptDir,
+            Middleware::Csv => Middleware::CsvDir,
+            Middleware::JsonModel | Middleware::Text => Middleware::Dir,
+            _ => middleware,
+        }
+    }
+
+    if middleware == Middleware::Rbxm {
+        middleware = match env::var(DEBUG_MODEL_FORMAT_VAR) {
+            Ok(value) if value == "1" => Middleware::Rbxmx,
+            Ok(value) if value == "2" => Middleware::JsonModel,
+            _ => Middleware::Rbxm,
+        }
+    }
+
+    middleware
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(deny_unknown_fields, rename_all = "camelCase")]
+pub struct SyncbackRules {
+    /// A list of subtrees in a file that will be ignored by Syncback.
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    ignore_trees: Vec<String>,
+    /// A list of patterns to check against the path an Instance would serialize
+    /// to. If a path matches one of these, the Instance won't be syncbacked.
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    ignore_paths: Vec<String>,
+    /// A map of classes to properties to ignore for that class when doing
+    /// syncback.
+    #[serde(default, skip_serializing_if = "HashMap::is_empty")]
+    ignore_properties: HashMap<String, Vec<String>>,
+    /// Whether or not the `CurrentCamera` of `Workspace` is included in the
+    /// syncback or not. Defaults to `false`.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    sync_current_camera: Option<bool>,
+    /// Whether or not to sync properties that cannot be modified via scripts.
+    /// Defaults to `false`.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    sync_unscriptable: Option<bool>,
```

Comment:
Defaulting to false might not be a great idea. This means users have to care about whether the properties they want to write back are scriptable or not, and they won't find out about it until they want to write back a property that happens to be unscriptable. This will seem like a bug to them! 

I understand that the purpose is to prevent the Rojo plugin from displaying warnings when syncing a project generated by syncback, but this doesn't seem like a pressing issue because most syncback workflows will always start with a `rojo build`, so the Rojo plugin won't attempt to set the unscriptable properties.

Reply:
You nailed why I had this default to false. The workflow I had in mind is new users who are doing one time conversions of their games to Rojo, but I think it might actually be bad behavior for them too. I'll change it to default to `true`; good call.